core/cochran.c

```diff
@@ -791,10 +791,11 @@ static void cochran_parse_dive(const unsigned char *decode, unsigned mod,
 	free(buf);
 }
 
-int try_to_open_cochran(const char *filename, struct memblock *mem, struct dive_table *table, struct trip_table *trips)
+int try_to_open_cochran(const char *filename, struct memblock *mem, struct dive_table *table, struct trip_table *trips, struct dive_site_table *sites)
```

Comment:
Saying that I put any thought into these function arguments would be a lie. Nevertheless, the fundamental difference is that the former functions are dive-site only and therefore `table` is clear from context. The latter are parser-functions which already take two other table-arguments (viz. dive and trip).

I changed the dive-site functions to take a `ds_table`. That was so painful (I'm very clumsy with git) that I'd prefer changing the parser functions with an additional commit. Note that for consistency, we'd also have to change the `trips` parameter to `trip_table`. Which will cause lgtm.com to complain, because `trip_table` is the name of a global variable.

Anyway, I entertain the though to replace the dive-, trip- and site-table triple by a
```
struct dive_log {
        struct dive_table dives;
        struct trip_table trips;
        struct dive_site_table sites;
};
```
 That would simplify these function signatures *and* replace three global variables by one.